lib/common/button/x_button.dart

```diff
@@ -0,0 +1,48 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_devtools/common/progress_indicator/x_circular_progress_indicator.dart';
+import 'package:flutter_devtools/presentation/theme/colors/variations/app_colors.dart';
+
+part 'x_button_state.dart';
+
+class XButton extends StatelessWidget {
+  const XButton({
+    required this.child,
+    required this.onPressed,
+    this.state = XButtonState.ENABLED,
+    Key? key,
+  }) : super(key: key);
+
+  final Widget child;
+  final XButtonState state;
+  final VoidCallback onPressed;
+
+  @override
+  Widget build(BuildContext context) {
+    return ElevatedButton(
+      onPressed: state.isDisabled
```

Comment:
here you could externalize to a getter like `VoidCallback? get _onPressed => ...`

Reply:
then use a switch to each case